src/matrixrtc/MatrixRTCSessionManager.ts

```diff
@@ -95,6 +98,18 @@ export class MatrixRTCSessionManager extends TypedEventEmitter<MatrixRTCSessionM
         return this.roomSessions.get(room.roomId)!;
     }
 
+    private onTimeline = (event: MatrixEvent): void => {
+        if (event.getType() !== EventType.CallEncryptionKeysPrefix) return;
+
+        const room = this.client.getRoom(event.getRoomId());
+        if (!room) {
+            logger.error(`Got room state event for unknown room ${event.getRoomId()}!`);
+            return;
+        }
+
+        this.getRoomSession(room).onCallEncryption(event);
+    };
```

Comment:
Non-blocking, but I'm not convinced that this callback belongs in the session manager class. I think it makes most sense if the session manager is only responsible for creating, ending, and storing sessions, so this feels better suited for the session class itself.

Reply:
Yeah, I think I would agree: I guess it was sticking to the pattern of all the events going into the manager and being dispatched from there.